packages/flutter_tools/lib/src/fuchsia/fuchsia_device.dart

```diff
@@ -442,6 +442,28 @@ class FuchsiaDevice extends Device {
     }
   }
 
+  @override
+  bool get supportsScreenshot => true;
+
+  @override
+  Future<void> takeScreenshot(File outputFile) async {
+    if (outputFile.basename.split('.').last != 'ppm') {
+      throw '${outputFile.path} must be a .ppm file';
+    }
+    final RunResult screencapResult = await shell('screencap > /tmp/screenshot.ppm');
+    if (screencapResult.exitCode != 0) {
+      throw 'Could not take a screenshot on device $name';
+    }
+    try {
+      final RunResult scpResult =  await scp('/tmp/screenshot.ppm', outputFile.path);
+      if (scpResult.exitCode != 0) {
+        throw 'Failed to copy screenshot from device';
+      }
+    } finally {
+      await shell('rm /tmp/screenshot.ppm');
```

Comment:
This might need to be wrapped in a `try {} catch {}` to ignore errors when failing to delete. The `catch` could do a `printError`.